File: Src2/Icm.Core/Color console/ColorString.vb
Namespace Icm.ColorConsole

    ''' <summary>
    ''' String with foreground and background colors.
    ''' </summary>
    ''' <remarks>
    ''' </remarks>
    Public Class ColorString

        Property ForegroundColor() As ConsoleColor?
        Property BackgroundColor() As ConsoleColor?
        Property [String]() As String

        Public Sub New(ByVal str As String, ByVal foreclr As ConsoleColor?, ByVal backclr As ConsoleColor?)
            ForegroundColor = foreclr
            BackgroundColor = backclr
            [String] = str
        End Sub

    End Class

End Namespace

File: Src/Icm.Core/ColorConsole/ColorString.cs
using System;

namespace Icm.ColorConsole
{

    /// <summary>
    /// String with foreground and background colors.
    /// </summary>
    /// <remarks>
    /// </remarks>
    public class ColorString
    {

        public ConsoleColor? ForegroundColor { get; }
        public ConsoleColor? BackgroundColor { get; }
        public string String { get; }

        public ColorString(string str, ConsoleColor? foreclr, ConsoleColor? backclr)
        {
            ForegroundColor = foreclr;
            BackgroundColor = backclr;
            String = str;
        }

    }

}
